go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -653,25 +658,17 @@ func (agent *ActionAgent) SlaveWasRestarted(ctx context.Context, parent *topodat
 	}
 	defer agent.unlock()
 
-	typeChanged := false
-
-	// Once this action completes, update authoritative tablet node first.
-	if _, err := agent.TopoServer.UpdateTabletFields(ctx, agent.TabletAlias, func(tablet *topodatapb.Tablet) error {
-		if tablet.Type == topodatapb.TabletType_MASTER {
-			tablet.Type = topodatapb.TabletType_REPLICA
-			typeChanged = true
-			return nil
-		}
-		return topo.NewError(topo.NoUpdateNeeded, agent.TabletAlias.String())
-	}); err != nil {
-		return err
-	}
-
-	if typeChanged {
-		if err := agent.refreshTablet(ctx, "SlaveWasRestarted"); err != nil {
+	if agent.Tablet().Type == topodatapb.TabletType_MASTER {
```

Comment:
I think here we can also just always call `ChangeType` after we fix it to use `NoUpdateNeeded`. The catch is we won't know whether anything changed. We could add a `changed?` return value to `ChangeType` if that would be generally useful, although the downside is it makes many other call sites uglier.

If this is the only case that needs to know if anything changed, we could also work around it by just always calling `refreshTablet`. In theory, that should itself be a no-op if nothing changed.

Reply:
This is another place where we _do_ need to check the tablet type first.
As far as `changed?` goes, the semantics of UpdateTabletFields is that if the tablet didn't change, then the return value is nil. We can maintain those semantics for ChangeType as well.

Reply:
Done.